src/org/ssgwt/client/ui/form/spinner/DurationSpinner.java

```diff
@@ -0,0 +1,754 @@
+    }
+
+    /**
+     * Adds a style name to the spinners
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @param style - The style to add
+     */
+    @Override
+    public void addStyleName(String style) {
+        hourSpinnerContainer.addStyleName(style);
+        minuteSpinnerContainer.addStyleName(style);
+    }
+
+    /**
+     * Removes a style name from the spinners
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @param style - The style to remove
+     */
+    @Override
+    public void removeStyleName(String style) {
+        hourSpinnerContainer.removeStyleName(style);
+        minuteSpinnerContainer.removeStyleName(style);
+    }
+
+    /**
+     * Getter for the boolean whether the fields is enabled
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @return whether this widget is enabled.
+     */
+    public boolean isEnabled() {
+        return hourSpinner.isEnabled();
+    }
+
+    /**
+     * Sets whether this widget is enabled.
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @param enabled - true to enable the widget, false to disable it
+     */
+    public void setEnabled(boolean enabled) {
+        hourSpinner.setEnabled(enabled);
+        hourValueBox.setEnabled(enabled);
+        minuteSpinner.setEnabled(enabled);
+        minuteValueBox.setEnabled(enabled);
+    }
+
+    /**
+     * Sets the text boxes to ready only or not.
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @param enabled - true to set the text boxes to read only, false to enable it
+     */
+    public void setTextBoxReadOnly(boolean enabled) {
+        hourValueBox.setReadOnly(enabled);
+        minuteValueBox.setReadOnly(enabled);
+    }
+
+    /**
+     * Add the valueChangeHandler to the value spinner
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @param handler - The handler to add
+     *
+     * @return HandlerRegistration
+     */
+    @Override
+    public HandlerRegistration addValueChangeHandler(ValueChangeHandler<Integer> handler) {
+        return addHandler(handler, ValueChangeEvent.getType());
+    }
+
+    /**
+     * Get the total minutes value from the spinners
+     *
+     * @author Ruan Naude <ruan.naude@a24group.com>
+     * @since 15 June 2015
+     *
+     * @return The total minutes value from the spinners
+     */
+    @Override
+    public Integer getValue() {
+        try {
+            int hours = Integer.valueOf(this.hourValueBox.getValue());
+            int minutes = Integer.valueOf(this.minuteValueBox.getValue());
+            return minutes + (hours * 60);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * Set the value of the spinners
+     *
```

Comment:
added this to the function above, since that is the one that will always fire the event